include/zephyr/sys/byteorder.h

```diff
@@ -719,4 +721,120 @@ static inline void sys_mem_swap(void *buf, size_t length)
 	}
 }
 
+/**
+ *  @brief Convert buffer from little-endian to host endianness.
+ */
+static inline void sys_le_to_cpu(void *buf, size_t length)
+{
+	if (!IS_ENABLED(CONFIG_LITTLE_ENDIAN)) {
```

Comment:
`if (IS_ENABLED(CONFIG_BIG_ENDIAN))` would be slightly simpler.

Reply:
Can you elaborate on why this is simpler? 

Wouldn't it become 

```c
static inline void sys_le_to_cpu(void *buf, size_t length)
{
	if (IS_ENABLED(CONFIG_LITTLE_ENDIAN)) {
		/* no-op */ 
	} else {
		sys_mem_swap(buf, length);
	}
}
```

Reply:
`CONFIG_LITTLE_ENDIAN` vs `CONFIG_BIG_ENDIAN`

Reply:
Lik3 @pdgendt pointed out, seems like you missed the LITTLE -> BIG in my proposal.

Reply:
I did - Sorry :)

Reply:
Updated to suggestion.